lua/sp.c

```diff
@@ -3227,6 +3227,14 @@ static int dbthread_join(Lua lua1)
         struct timespec ts;
         clock_gettime(CLOCK_REALTIME, &ts);
         ts.tv_sec += 1;
+        if (bdb_lock_desired(thedb->bdb_env)) {
+            int rc;
+            if ((rc = recover_deadlock(thedb->bdb_env, lt->sp->thd->sqlthd, NULL, 0)) !=
+                    0) {
+                luabb_error(lua1, lt->sp, "recover deadlock failed");
+                return -3;
```

Comment:
Can we look into why recover_deadlock could fail here, and what is best to do when it does?

Reply:
Discussed with Akshat- he is going to take care of this ..

Reply:
I will "take care" of this.